balboa-kafka/src/main/scala/com/socrata/balboa/kafka/Main.scala

```diff
@@ -0,0 +1,133 @@
+package com.socrata.balboa.kafka
+
+import java.io.{IOException, File, FileReader}
+import java.util.Properties
+import java.util.concurrent.Executors
+
+import com.socrata.balboa.kafka.BalboaConfig
+import com.socrata.balboa.kafka.consumer.{DSConsumer, TestConsumer}
+import com.socrata.balboa.metrics.config.Configuration
+import com.socrata.balboa.metrics.data.{DataStoreFactory, DataStore}
+import kafka.consumer.ConsumerConfig
+
+/**
+ * Created by Michael Hotan, michael.hotan@socrata.com on 2/17/15.
+ *
+ * Balboa Kafka Main Application:
+ *
+ * Entry point for Balboa Kafka Consumer.  This application starts a service that listens
+ * on a configurable list of ZooKeeper Host Ports.  This service currently listens for all published Tenant Metrics.
+ *
+ */
+object Main extends App {
+
+  /**
+   * Parse handles command line configuration.
+   */
+  val parser = new scopt.OptionParser[BalboaConfig]("balboa-kafka") {
+    opt[File]('f', "configFile") action { (x,c) => c.copy(configFile = x) } validate {
+      x => if (x.exists()) success else failure("Configuration does not exist") } text("Kafka configuration file")
+    opt[String]('z', "zookeepers") action { (x,c) => c.copy(zookeepers = x.trim)} validate { x => if (x.split(',').length
+      > 0) success else failure("Option --zookeepers must have 1 or more host:port names") } text("Comma separated " +
+      "list of zookeeper host:port")
+    opt[String]('g', "groupid") action { (x,c) => c.copy(groupId = x.trim) } text("Consumer group id")
+    opt[String]('t', "topic") action {(x,c) => c.copy(topic = x)} text("The topic to consume")
+    opt[Int]('n', "threads") action {(x,c) => c.copy(threads = x)} text("Number of threads to use") validate(x => if
+    (x > 0) success else failure("Must have positive number of threads"))
+  }
+
+  // Retrieve the default data store.
+  val ds: DataStore = DataStoreFactory.get
+
+  // Actually parse the command line argument
+  parser.parse(args, BalboaConfig()) match {
+    case Some(config) =>
+
+      val props: Properties = properties(config) match {
+        case Right(properties) => properties
+        case Left(message) => sys.error(s"Unable to create properties: $message")
+      }
+
+      // Kafka Consumer configuration.
+      val consumerConfig: ConsumerConfig = new ConsumerConfig(props)
+      val threads: Int = props.getProperty("balboa.kafka.topic.partitions", "1").toInt
+      val topic: String = props.getProperty("balboa.kafka.topic")
+
+      val cg: BalboaConsumerGroup = BalboaConsumerGroup.init(consumerConfig, topic)
+      BalboaConsumerGroup.getStreams(cg, threads) match {
+        case Some(streams) =>
+          // For each available stream create a consumer to ingest the stream.
+          val executor = Executors.newFixedThreadPool(threads)
+          streams.foreach(stream => executor.submit(DSConsumer(stream, DataStoreFactory.get())))
+        case None => Console.println("No Available Stream to print")
```

Comment:
Agreed misleading error message.  It should be pumped into a log as well.  The documentation for Kafka's "createMessageStreams" is a little ambiguous to what a None optional result means.  I feel like because they used Option it indicates a non error but no streams available for a given topic.   But not sure :-(